### ISSUE TITLE
logicalsel Rexexp selects incorrect values since 2.2.3

### ISSUE THREAD

**Issue by paulbrimicombe** [dperini/nwsapi#88]
In version 2.2.3, the [`logicalsel` regular expression](https://github.com/dperini/nwsapi/compare/410638fc214f9d8cd8a403af67f0d0ea852f13c0..c16b71d92a692b2fcd30149f4a5e54a8a841cbc2#diff-8081ec5523d50813a65b2ec303d10ae06a5ba508866607db3700779f9bb62f07L79-R81) changed from:

```
logicalsel: '(is|where|matches|not)\\x28\\s?([^()]*|[^\\x28]*\\x28[^\\x29]*\\x29)\\s?(?:\\x29|$)',
```

to 

```
logicalsel: '(is|where|matches|not)\\x28\\s?([^\\[]*\\[[^\\]]*\\]|[^\\(]*\\([^\\)]*\\)|[^()]*)\\s?(?:\\x29|$)',
```

When matching a selector of the form:

[EDIT: this example was incorrect in the original bug report but should now be correct — apologies for the mistake!]

```css
div:not(.main) a:not([class=unstyled-link]) {
    text-decoration: underline;
}
```

The old version would match `not(.main)` and `not([class=unstyled-link])` separately but the new version matches the whole of `not(.unstyled-links) a:not([class=unstyled-link)]` as a single capture group. The `[]` in the second selector appears to be the problem — if I rewrite the second part of the selector as `a:not(.unstyled-link)` everything works as expected.

**Comment by dperini** [comment]
@paulbrimicombe 
it looks like the latest change I did to the Regular Expression solves for this selector and others too.
Have you tried latest Github nwsapi commits ?

**Comment by paulbrimicombe** [comment]
> @paulbrimicombe it looks like the latest change I did to the Regular Expression solves for this selector and others too. Have you tried latest Github nwsapi commits ?

@dperini the code in `master` definitely _looks_ correct (comparing the matches made by the old and new regular expressions) but without a pre-release published to NPM I wouldn't be able to tell for sure!

**Comment by dperini** [comment]
@paulbrimicombe @ALL
See issue #89 and commit 01216f85b9989ce0f1d5b0ee5ce28bb88ea8aaa3 for a new regular expression that should fix remaining failure cases.

**Comment by paulbrimicombe** [comment]
Hi @dperini , I've just checked v 2.2.5 and that looks good for us. It's harder to tell with the newer version because it's unreleased and I'd have to go in and hack `node_modules` to test it. Any chance of a pre-release published to npm?

**Comment by dperini** [comment]
@paulbrimicombe 
I need to release more fixes in 2.2.6 but it will be done soon.

**Comment by dperini** [comment]
@paulbrimicombe 
more fixes were committed to the "nwsapi" repository.
Keep trying and report any issue you might find.
Thank you for your feedback
